processor/events_response.go

```diff
@@ -0,0 +1,75 @@
+package processor
+
+import (
+	"time"
+
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/processor/transformer"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/samber/lo"
+)
+
+func (proc *Handle) getDroppedJobs(response transformer.Response, eventsToTransform []transformer.TransformerEvent) []*jobsdb.JobT {
+	// each messageID is one event when sending to the transformer
+	inputMessageIDs := lo.Map(eventsToTransform, func(e transformer.TransformerEvent, _ int) string {
+		return e.Metadata.MessageID
```

Comment:
I am confused why processor's pipeline operates on the assumption that messageIDs are unique. Dedup is a feature that can be turned off, right? Am I missing some other important fact with respect to message ids and the processor's pipeline?

Reply:
Yeah that is correct but dedup ideally should never be turned off

Also for many of the downstream destinations like Warehouse they do operate under the assumption that the primary key is the messageID which is unique